### ISSUE TITLE
-prefix-free disables media query in IE9

### ISSUE THREAD

**Issue by sussexrick** [LeaVerou/prefixfree#31]
In IE9 with a linked stylesheet using the media query "only screen and (min-width: 500px)", -prefix-free disables that stylesheet when it loads. Resizing the browser causes the stylesheet to be re-applied.

Test case: http://eastsussex.gov.uk/testcase/linked.html

**Comment by jaygeorge** [comment]
I have this issue as well - any ideas of a fix?

**Comment by LeaVerou** [comment]
Confirmed, sorry for the delay.

This is super odd. It seems the &lt;link> isn't even replaced with a &lt;style> element, XHR just hangs there. The media attribute isn't even removed or anything, so no idea why it doesn't work. Will investigate.

**Comment by curiouz** [comment]
I have this problem too!

**Comment by sussexrick** [comment]
Updated link for test case: http://eastsussex.gov.uk/testcases/prefix-free-ie9/noprefix.html

**Comment by grantmccall** [comment]
Any update on a fix for this? Thanks.

**Comment by dziudek** [comment]
I've the same problem with IE9 and it seems that the solution is simple but mysterious for me ;)

In the StyleFix I've added after lines:

``` js
var style = document.createElement('style');
style.textContent = css;
style.media = link.media;
style.disabled = link.disabled;
style.setAttribute('data-href', link.getAttribute('href'));

parent.insertBefore(style, link);
parent.removeChild(link);
```

line:

``` js
style.media = link.media;
```

and... it works fine for me in IE9 and in Chrome but the best part is fact that code:

``` js
var style = document.createElement('style');
style.textContent = css;
style.disabled = link.disabled;
style.setAttribute('data-href', link.getAttribute('href'));

parent.insertBefore(style, link);
parent.removeChild(link);

style.media = link.media;
```

is working fine in IE9 but not in the Chrome ^^ That's why my fix contains duplicated line:

``` js
style.media = link.media;
```

@LeaVerou - could you look and analyze my solution if it is accurate? I've checked it also in case when the browser window is small at start.

**Comment by LeaVerou** [comment]
Interesting. Maybe `style.setAttribute('media', link.getAttribute('media') || 'all')` solves it regardless of position in the code?

**Comment by dziudek** [comment]
Unfortunately it doesn't work placed not in the first nor in the second position :/ It seems that the declaration of the style.media has to be placed in two lines.

**Comment by LeaVerou** [comment]
Could you elaborate? It doesn't work in either browser?

**Comment by dziudek** [comment]
It doesn't work for me on the IE9 on the first position and for the Chrome in the second position - of course when this line is placed in the both positions then it works fine.

**Comment by LeaVerou** [comment]
This is insane. :( But, since it's the only viable solution right now, lets do it. 
Since you found the workaround, can you please submit a pull request with it? Don't forget to:
1. Add a short comment that explains why the line is duplicated
2. Update the minified version too

Thanks!

**Comment by dziudek** [comment]
@LeaVerou - done :) For me it is also amazing, but it is IE ;)

**Comment by gausarts** [comment]
Tracking. Same issue.

**Comment by mkantor** [comment]
For posterity, this was fixed in [pull request #73](http://github.com/LeaVerou/prefixfree/pull/73).

**Comment by LeaVerou** [comment]
Closing then.
